File: Changwhi/is-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        sChars = {}
        tChars = {}

        for i in range(len(s)):
            currentSValue = sChars.get(s[i], 0)
            sChars[s[i]] = currentSValue + 1

            currentTvalue = tChars.get(t[i], 0)
            tChars[t[i]] = currentTvalue + 1

        return sChars == tChars

File: 2024/hj/IsAnagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) return false;

        unordered_map<char, int> um;

        for (int i = 0; i < s.size(); i++) {
            um[s[i]]++;
            um[t[i]]--;
        }

        for (auto it = um.begin(); it != um.end(); it++)
            if (it->second != 0) return false;

        return true;
    }
};
